kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/utils/ExpressionTestUtils.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.utils
+
+import io.delta.kernel.expressions.{Column, Expression, Literal, Predicate}
+
+/** Useful helper functions for creating expressions in tests */
+trait ExpressionTestUtils {
+
+  def equals(e1: Expression, e2: Expression): Predicate = {
+    new Predicate("=", e1, e2)
+  }
+
+  def lessThan(e1: Expression, e2: Expression): Predicate = {
+    new Predicate("<", e1, e2)
+  }
+
+  def greaterThan(e1: Expression, e2: Expression): Predicate = {
+    new Predicate(">", e1, e2)
+  }
+
+  def greaterThanOrEqual(e1: Expression, e2: Expression): Predicate = {
+    new Predicate(">=", e1, e2)
+  }
+
+  def lessThanOrEqual(e1: Expression, e2: Expression): Predicate = {
+    new Predicate("<=", e1, e2)
+  }
+
+  def not(pred: Predicate): Predicate = {
+    new Predicate("NOT", pred)
+  }
+
+  def isNotNull(e1: Expression): Predicate = {
+    new Predicate("IS_NOT_NULL", e1)
+  }
+
+  def col(name: String): Column = new Column(name)
+
+  def nestedCol(name: String): Column = {
+    new Column(name.split("\\."))
+  }
+
+  /* ---------- NOT-YET SUPPORTED EXPRESSIONS ----------- */
+
+  def nullSafeEquals(e1: Expression, e2: Expression): Predicate = {
+    new Predicate("<=>", e1, e2)
```

Comment:
maybe comment this out, and throw an exception if they are invoked?

or, why write code that can't be used, instead of just comments?

Reply:
They will throw an exception if they are tried to be evaluated.

See as to why they are in the tests https://github.com/delta-io/delta/pull/2433#discussion_r1443646582. For now, unsupported expressions should just be ignored in the data skipping filter generation (and will be returned to the connector as part of the remaining predicate to evaluate.)

Reply:
I can add a comment here explaining this reasoning